.github/workflows/check-plugin.yml

```diff
@@ -42,7 +42,7 @@ jobs:
       - name: Install dependencies
         run: |
           cd ${{ inputs.plugin }}
-          pip install ".[test]"
+          pip install . -r test_requirements.txt  # TODO(deepyaman): Define `test` extra and `pip install .[test]`
```

Comment:
I'm wondering if we could keep this one? Hence retain the `test` dependencies in the project metadata file (will be `setup.py` in this PR), rather than going back to a separate `test_requirements.txt`.

Reply:
Yes, it seems we need to modify our approach. We should not only revert but transfer the `test_requirements.txt` to `setup.py` instead of `pyproject.toml`. I can't simply revert changes for only kedro-datasets, as it would lead to inconsistencies with test_requirements. Currently, I've caused a disruption in the end-to-end test with RTD, since it was previously assumed that we could use ".[test]".